lnst/Recipes/ENRT/MeasurementGenerators/FlowMeasurementGenerator.py

```diff
@@ -52,188 +19,18 @@ class FlowMeasurementGenerator(BaseMeasurementGenerator):
-
-    def generate_perf_endpoints(self, config):
-        """Generator for perf endpoints
-
-        To be overridden by a derived class.
-
-        :return: list of device pairs
-        :rtype: List[Tuple[:any:`Device`, :any:`Device`]]
-        """
-        return []
-
-    def _create_perf_flows(
-        self,
-        perf_test,
-        client_nic,
-        client_bind,
-        server_nic,
-        server_bind,
-        msg_size,
-    ) -> List[PerfFlow]:
-        flows = []
-        port_offset=12000
-        for i in range(self.params.perf_parallel_processes):
-            flows.append(
-                self._create_perf_flow(
-                    perf_test,
-                    client_nic,
-                    port_offset + i,
-                    server_nic,
-                    server_bind,
-                    port_offset + i,
-                    msg_size,
-                    self._cpupin_based_on_policy(i),
-                )
-            )
-
-        return flows
-
-    def _cpupin_based_on_policy(self, process_no=None):
-        if process_no is None:
-            return None
-
-        try:
-            cpus = self.params.perf_tool_cpu
-        except:
-            return None
-
-        try:
-            policy = self.params.perf_tool_cpu_policy
-        except:
-            return cpus
+    def generator_cpupin(self, process_no):
+        cpus = self._cpupin_based_on_policy(
+            process_no,
+            self.params.get('perf_tool_cpu', None),
+            self.params.get('perf_tool_cpu_policy', None),
+        )
 
-        if policy == 'round-robin':
-            return [cpus[process_no % len(cpus)]]
-        elif policy == 'all':
-            return cpus
-        else:
-            raise Exception(f'Unknown perf_tool_cpu_policy {policy}')
+        return cpus
 
-    def _create_perf_flow(
-        self,
-        perf_test,
-        client_nic,
-        client_bind,
-        client_port,
-        server_nic,
-        server_bind,
-        server_port,
-        msg_size,
-        cpupin,
-    ) -> PerfFlow:
-        """
-        Wrapper to create a PerfFlow. Mixins that want to change this behavior (for example, to reverse the direction)
-        can override this method as an alternative to overriding :any:`generate_flow_combinations`
-        """
-        return PerfFlow(
-            type=perf_test,
-            generator=client_nic.netns,
-            generator_bind=client_bind,
-            generator_nic=client_nic,
-            generator_port=client_port,
-            receiver=server_nic.netns,
-            receiver_bind=server_bind,
-            receiver_nic=server_nic,
-            receiver_port=server_port,
-            msg_size=msg_size,
-            duration=self.params.perf_duration,
-            warmup_duration=self.params.perf_warmup_duration,
-            parallel_streams=self.params.perf_parallel_streams,
-            cpupin=cpupin,
-        )
+    def receiver_cpupin(self, process_no):
```

Comment:
shouldn't this by default be the same as for the generator?
```
        cpus = self._cpupin_based_on_policy(
            process_no,
            self.params.get('perf_tool_cpu', None),
            self.params.get('perf_tool_cpu_policy', None),
        )
```

Reply:
ah right, i think i misread the code as just using `self._cpupin_based_on_policy()` again and not just reusing the `generator` property and got confused... this is ok. resolving.